operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -115,12 +114,13 @@ public void delete(P primary, Context context) {
     resource.ifPresent(r -> client.resource(r).delete());
   }
 
+  protected abstract Class<R> resourceType();
```

Comment:
This makes things harder for the Quarkus extension because it will now need to check more possible cases.

Reply:
What exactly?